packages/@n8n/ai-workflow-builder.ee/src/workflow-builder-agent.ts

```diff
@@ -78,15 +78,47 @@ export function shouldModifyState(
 		return 'create_workflow_name';
 	}
 
+	const workflowContextToAppend = getWorkflowContext(state);
+
 	// Check if we should auto-compact based on token count
-	const estimatedTokens = estimateTokenCountFromMessages(messages);
+	const estimatedTokens = estimateTokenCountFromMessages([
+		...messages,
+		// appended later to last message
+		new HumanMessage(workflowContextToAppend),
+	]);
```

Comment:
We might need to account for tool definitions tokens here as well. We can add this later though